### PR TITLE
testutil: make testing packages public

### PR DESCRIPTION
**Pull request by SamWhited**
**- What I did**

Moved `/internal/test` and `/internal/testutil` into `/testutil` so that it can be used from within buildkit.

**- How I did it**

Moved it and sed-ed all of the import paths and package names.

**- How to verify it**

Hopefully CI should pass and git should show that most files are just moved and not modified heavily.

**- Description for the changelog**

The testutil package is now public and can be used to spawn docker daemons for testing.

**- A picture of a cute animal (not mandatory but encouraged)**

Gopher

### CODE REVIEW

**Comment by tiborvass** [comment]
Thanks @SamWhited! For posterity I like to put in the commit message at least, the exact sed command used.

**Comment by SamWhited** [comment]
Sure. It was something along the lines of:

```                                                                                                                                                                                                     
mv internal/test testutil                                                                                                                                                                               
pushd testutil/; grep -IRl "package test" | xargs -I '{}' sed -i -e 's|package test|package testutil|g' {}; popd                                                                                        
mv internal/testutil/*.go testutil/ && rm -rf internal/                                                                                                                                                 
grep -IRl "github.com\/docker\/docker\/internal\/test" | xargs -I '{}' sed -i -e 's|github.com/docker/docker/internal/test|github.com/docker/docker/test|g' {}                                          
```

That may not be exact, I copied a few lines out of my bash history. I'll throw this in the commit message after CI finishes (I don't want to interrupt it)

File: testutil/stringutils.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package testutil // import "github.com/docker/docker/internal/testutil"
```
Comment (thaJeztah) [inline testutil/stringutils.go]: curious; why was this removed?

File: testutil/stringutils_test.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package testutil // import "github.com/docker/docker/internal/testutil"
```
Comment (thaJeztah) [inline testutil/stringutils_test.go]: same here

File: testutil/environment/clean.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package environment // import "github.com/docker/docker/internal/test/environment"
+package environment
```
Comment (thaJeztah) [inline testutil/environment/clean.go]: why was this removed?

File: testutil/daemon/daemon_unix.go [hunk 4]
```diff
@@ -1,20 +1,20 @@
 // +build !windows
 
-package daemon // import "github.com/docker/docker/internal/test/daemon"
```
Comment (thaJeztah) [inline testutil/daemon/daemon_unix.go]: here as well

File: testutil/daemon/daemon.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package daemon // import "github.com/docker/docker/internal/test/daemon"
```
Comment (thaJeztah) [inline testutil/daemon/daemon.go]: and here

File: testutil/helpers.go [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package testutil // import "github.com/docker/docker/internal/testutil"
```
Comment (thaJeztah) [inline testutil/helpers.go]: and here

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: testutil/environment/clean.go (hunk 3 above)
Comment (SamWhited) [reply testutil/environment/clean.go]: Only one of these is needed per package; since it doesn't do anything having them in multiple files and I was having to touch the line anyways I figured I'd remove them so that it wouldn't be a part of the diff in the future if this package is ever moved again.

Comment (thaJeztah) [reply testutil/environment/clean.go]: Removing them adds more code-churn (trying to avoid that); We should probably have a validate step to make sure packages have these comments; would it make it easier  to have them in all files (even if redundant)?

Comment (SamWhited) [reply testutil/environment/clean.go]: Although note that I didn't audit every package to see if it needed the extra import lines removed, I just happened to notice it in one or two and figured incremental cleanup would be good :)

Comment (SamWhited) [reply testutil/environment/clean.go]: > Removing them adds more code-churn (trying to avoid that)

Fair enough, although note that I had to touch these lines anyways to update them, so I just thought if we were updating them we might as well remove them and save code churn next time (if there is a next time, hopefully not).

> We should probably have a validate step to make sure packages have these comments; would it make it easier to have them in all files (even if redundant)?

I don't think so, it just seems more likely that a typo could break things because we end up with conflicting import comments to me, but I don't think it's super important either way.

Let me know if you'd like me to add them back here, I still need to figure out why CI is failing so I can do this before pushing the next update.

**Review by tiborvass — APPROVED** [review]

**Comment by SamWhited** [comment]
Two things still failing in CI, one appears to be the Windows issue that I believe is being worked on in another PR, the other doesn't appear to be related but I'm not sure why it's failing now. Will dig in as soon as I'm done with some of the buildkit changes.

**Comment by SamWhited** [comment]
Looks like running goimports makes a lot of changes, fixing only the ones that were the result of the package rename to try and make things work for now.

**Comment by SamWhited** [comment]
Remaining test failures are the windows failures which I think are failing in general right now and aren't related. PTAL. Thanks!

**Comment by SamWhited** [comment]
Running tests appears to be failing for different reasons now, looking into it to see if it's my fault or not. They do appear to be failing (sometimes?) on master too, so maybe it's just flaky. If anyone is familiar with windows tests and what the state of testing on master is right now your input would be appreciated. Thanks! /cc @tiborvass, @thaJeztah

File: testutil/environment/clean.go (hunk 3 above)
Comment (SamWhited) [reply testutil/environment/clean.go]: Ping @thaJeztah on this last question; would you prefer that I revert this change? Thanks!

**Review by tiborvass — APPROVED** [review]

**Comment by SamWhited** [comment]
Rebased on top of Tibor's testing change. Waiting on CI to see if I missed anything or broken an import.

**Review by tiborvass — APPROVED** [review]

File: testutil/environment/clean.go (hunk 3 above)
Comment (thaJeztah) [reply testutil/environment/clean.go]: oh! sorry, missed your last comment.

Yes, I'd prefer reverting it.

Comment (thaJeztah) [reply testutil/environment/clean.go]: actually; let's work on a test to make sure every package has the comment set. I wonder if that also would help with some packages still showing up in the wrong location on godoc.org https://godoc.org/github.com/moby/moby

**Review by thaJeztah — APPROVED** [review]
LGTM

File: testutil/environment/clean.go (hunk 3 above)
Comment (SamWhited) [reply testutil/environment/clean.go]: As long as it's set once it should show up in the right place on godoc, it doesn't need to be set in every file though.

**Comment by SamWhited** [comment]
Pushed a change to add import lines back as requested by @thaJeztah; PTAL, thanks!

**Review by thaJeztah — APPROVED** [review]
LGTM

let's create a tracking issue to validate that each package has a (correct) import comment

**Comment by SamWhited** [comment]
@thaJeztah done, see #39898

**Comment by thaJeztah** [comment]
Oh, sorry; this needs a rebase now; I see https://github.com/moby/moby/pull/39886 was merged, which caused a conflict

**Comment by thaJeztah** [comment]
Only failure was on Windows RS1, and a known flaky test (on slow platforms?) https://github.com/moby/moby/issues/38521; 

```
     --- FAIL: Test/DockerSuite/TestStartReturnCorrectExitCode (7.62s)
         docker_cli_start_test.go:198: assertion failed: 125 (exitCode int) != 12 (int): out: time="2019-09-11T00:09:35Z" level=error msg="error waiting for container: EOF" 
```

**Comment by SamWhited** [comment]
@thaJeztah rebased

**Comment by thaJeztah** [comment]
Windows RS1 failure is a Jenkins issue; https://ci.docker.com/public/job/moby/job/PR-39829/17/execution/node/72/log/

```
 azwin-2-102ef0 was marked offline: Connection was broken: java.util.concurrent.TimeoutException: Ping started at 1568206049555 hasn't completed by 1568206289556
 	at hudson.remoting.PingThread.ping(PingThread.java:134)
 	at hudson.remoting.PingThread.run(PingThread.java:90)
```

Gonna ignore that one
